File: HTLCs/Starknet/scripts/call_api.py

import requests
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

API_URL = os.getenv("API_URL")
API_KEY = os.getenv("API_KEY")

def call_api(usdc_amount, btc_recipient, rootstock_recipient, flow_recipient):
    try:
        response = requests.post(
            API_URL,
            json={
                "apiKey": API_KEY,
                "usdcAmount": usdc_amount,
                "btcRecipient": btc_recipient,
                "rootstockRecipient": rootstock_recipient,
                "flowRecipient": flow_recipient
            }
        )
        print(f"API call result: {response.status_code}, {response.json()}")
    except Exception as e:
        print(f"Error in API call: {e}")

File: contracts/htlc/evm/scripts/callAPI.js
const axios = require("axios");
require("dotenv").config();

async function callAPI(btcRecipient, btcAmount) {
  try {
    // Load API key from environment variables
    const apiUrl = "https://127.0.0.1:5000/transfer_tokens"; // New API URL as per your request

    // Construct the query parameters for the API call
    const queryParams = `?recipient=${btcRecipient}&amount=${btcAmount}`;

    // Send a GET request to the API with query parameters
    const response = await axios.get(apiUrl + queryParams);

    console.log(`API call successful: ${response.data}`);
  } catch (error) {
    console.error(`API call failed: ${error.message}`);
  }
}

module.exports = callAPI;
